### ISSUE TITLE
[@types/sinon] `withArgs` does not accept sinon matchers.

### ISSUE THREAD

**Issue by stroiman** [DefinitelyTyped/DefinitelyTyped#31262]
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems (5.0.7)
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript (3.2.2)
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @mrbigdog2u @rationull @lumaxis @nicojs @43081j @joshuakgoldberg @gjednaszewski @johnjesse

You should be able to pass the output of `sinon.match` as input to `withArgs` - however that doesn't compile in the example below.

```javascript
const foo = {
    bar: (x: string) => x === "bar"
};

// No error 
sinon.stub(foo, "bar").withArgs("bar").returns(true); 

// error TS2345: Argument of type 'SinonMatcher' is not assignable to parameter of type 'string'.
sinon.stub(foo, "bar").withArgs(sinon.match(x => x === "baz")).returns(true);
```

**Comment by 43081j** [comment]
The fix for this is in #30856 afaik.

Its basically the fix-all PR we're all waiting on but maintainers of DT shuffled recently so its going slowly...

**Comment by stroiman** [comment]
Ok, I guess this issues can just be closed then.
